src/rgw/rgw_main.cc

```diff
@@ -403,10 +403,7 @@ int main(int argc, const char **argv)
 
       fe = new RGWFCGXFrontend(fcgi_pe, config);
     } else if (framework == "civetweb" || framework == "mongoose") {
-      int port;
-      config->get_val("port", 80, &port);
-
-      RGWProcessEnv env = { store, &rest, olog, port };
+      RGWProcessEnv env = { store, &rest, olog, 0 };
```

Comment:
We could either 
1. parse port=".." the way civetweb does, ie. allow ip:port combo, r, s params that civetweb allows etc.

2.  alternatively, we can set override the env.port after initializing port from civetweb and getting the value of ports civetweb parses using the `mg_get_ports` api ?; ie something like this in rgw_civetweb_frontend:

```c++
(assume ports = new int [num ports] etc)
size_t len = mg_get_ports(ctx, num_ports, ports, ssl_ports)
if (len){
    env.port = ports[0] ;}
``` 
etc 
?